stats/stats.go

```diff
@@ -0,0 +1,688 @@
+package stats
```

Comment:
- File is very huge lets cut short move all utility functions into separate file in same package something like stats_utils.go.

- ArtifactoryStats struct{} should have methods like
    - WriteTable
    - WriteText
    - WriteJson
probably think about creating an interface StatsWriter{} with above methods.

- All formatting based on flag should go to above methods.

- Create a method called filter and apply filters here basically your Writers should call filter. and filter should call the right API's 

- If the filter has only releaseBundle then there is no need to make other API calls.